Python/pystate.c

```diff
@@ -268,7 +297,10 @@ init_interpreter(PyInterpreterState *interp,
                  PyThread_type_lock pending_lock)
 {
     if (interp->_initialized) {
-        Py_FatalError("interpreter already initialized");
+        assert(interp->_preallocated.initialized);
```

Comment:
What is the justification for this change in behavior?

Reply:
Before, the interpreter was dynamically allocated and `_initialized` was always false.  Now the main interpreter is statically allocated and may be re-used if `Py_Initialize()` is called after `Py_Finalize()`.

Reply:
I took care of this a different way and have changed back to the `Py_FatalError()`.